src/main/java/com/google/ehub/servlets/ItemPageServlet.java

```diff
@@ -68,14 +75,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       System.err.println("Can't parse itemId to a long");
       return;
     }
-    String comment = request.getParameter(CommentDataManager.COMMENT_PROPERTY_KEY);
-    if (comment == null) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Comment was not entered.");
-      return;
+    if (!userService.isUserLoggedIn()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "User must logged in");
     } else {
-      long timestampMillis = System.currentTimeMillis();
-      CommentDataManager comments = new CommentDataManager();
-      comments.addItemComment(itemId, comment, timestampMillis);
+      String email = userService.getCurrentUser().getEmail();
+      String username = profileData.getUserProfile(email).getUsername();
+
+      String comment = request.getParameter(CommentDataManager.COMMENT_PROPERTY_KEY);
+      if (comment == null) {
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Comment was not entered.");
+        return;
+      } else {
+        long timestampMillis = System.currentTimeMillis();
+        CommentDataManager comments = new CommentDataManager();
+        comments.addItemComment(itemId, comment, timestampMillis, username);
```

Comment:
You can still store the user's email in the datastore. I think Rodrigo was suggesting to just display the username on the JS client, but storing the email is safer because it won't change.